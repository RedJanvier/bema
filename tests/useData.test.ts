import './__setup'

test('bema-level useData provides data to the context', async () => {
  bema.useData({ a: 1 })
  const ga = bema.group('a')
  ga.case().run(f1)
  await run(bema)
  expect(f1.mock.calls).toMatchSnapshot()
})

test('group-level useData provides data to the context', async () => {
  const ga = bema.group('a').useData({ a: 1 })
  ga.case().run(f1)
  await run(bema)
  expect(f1.mock.calls).toMatchSnapshot()
})
