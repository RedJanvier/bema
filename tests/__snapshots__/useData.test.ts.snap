// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bema-level useData provides data to the context 1`] = `
Array [
  Array [
    Object {
      "$info": Object {
        "parameters": Object {},
      },
      "a": 1,
    },
  ],
]
`;

exports[`group-level useData provides data to the context 1`] = `
Array [
  Array [
    Object {
      "$info": Object {
        "parameters": Object {},
      },
      "a": 1,
    },
  ],
]
`;
